README.md
# Freemius License Installer

* Author:            Andy Fragen
* License:           MIT
* Requires at least: 5.9
* Requires PHP:      7.2
* Tested up to:      trunk
* Stable version:    main
* Donate link:       <https://thefragens.com/git-updater-donate>

Add Freemius licensing from `.env` file. Uses https://github.com/squarecandy/freemius-auto-activation plugin.

## Description

Activating this plugin loads your license data from a `.env` file. You will need to create this `.env` file for the plugin to function. A PHP Fatal Exception error on activation will be thrown if the `.env` file is not found in the correct path.

Refer to `.env.example` for the format of the `.env` file.

The `.env` file will need to be created in the plugin root directory. This is a tradeoff as it is less secure but provides for an easier installation for you and your users.

The shortcode is unique to each Freemius integrated plugin or theme. Ask the plugin developer for this shortcode. It is usually evident in the Freemius integration code.

It will also automatically install and activate the [Freemius Auto Activation](https://github.com/squarecandy/freemius-auto-activation) plugin.

After running, the licenses with be loaded in the WordPress database. You may still be asked to opt in to telemetry data. Please agree to this instead of skipping.

After running this plugin, it may be deactivated and/or deleted. Upon deactivation, the Freemius Auto Activation plugin will be deleted.


CHANGES.md
[unreleased]

#### 1.1.0 / 2023-05-30
* composer update

#### 1.0.0 / 2023-04-14
* add .env to plugin root, tradeoff less secure but easier to distribute
* update requirements

#### 0.8.0 / 2023-04-04
* composer update

#### 0.7.0 / 2023-03-30
* composer update

#### 0.6.0 / 2023-01-31
* fix composer issue

#### 0.5.0 / 2023-01-19
* composer update

#### 0.4.0 / 2023-01-02
* composer update

#### 0.3.0 / 2022-09-24
* ensure `$wp_filesystem` is loaded

#### 0.2.2 / 2022-07-15
* delete Freemius Auto Activation plugin on deactivation
* composer update

#### 0.2.1 / 2022-05-25
* composer update

#### 0.2.0 / 2022-02-07
* use `.env` file for license data
* use `vlucas/phpdotenv` to read `.env` data
* rename plugin

#### 0.1.0 / 2022-02-05
* initial pass
* use wp-dependency-installer for `squarecandy/freemius-auto-activation`


git-updater-licensing.php
<?php
/**
 * Git Updater Licensing.
 *
 * @package git-updater-licensing
 * @author  Andy Fragen
 * @link    https://github.com/afragen/git-updater-licensing
 *
 * Plugin Name:       Git Updater Licensing
 * Plugin URI:        https://github.com/afragen/git-updater-licensing
 * Description:       Add Git Updater Freemius licensing.
 * Version:           0.1.0
 * Author:            Andy Fragen
 * License:           MIT
 * Requires at least: 5.2
 * Requires PHP:      7.1
 * GitHub Plugin URI: https://github.com/afragen/git-updater-licensing
 * Primary Branch:    main
 */

namespace Fragen\Git_Updater_Licensing;

/*
 * Exit if called directly.
 * PHP version check and exit.
 */
if ( ! defined( 'WPINC' ) ) {
	die;
}

require_once __DIR__ . '/vendor/autoload.php';


add_action(
	'plugins_loaded',
	function() {
		$config = [
			[
				'name'     => 'Freemius Auto Activation',
				'host'     => 'github',
				'slug'     => 'freemius-auto-activation/freemius-auto-activation.php',
				'uri'      => 'https://github.com/squarecandy/freemius-auto-activation',
				'branch'   => 'main',
				'required' => true,
			],
		];
		\WP_Dependency_Installer::instance( __DIR__ )->register( $config )->run();
	}
);

define( 'FS_SHORTCODES', [ 'gup_fs', 'gua_fs', 'gru_fs' ] );
define( 'WP__GUP_FS__LICENSE_KEY', 'sk_8f9W_vJiTyZ0.&_3@tP2-MwpQw9Zt' );
define( 'WP__GUA_FS__LICENSE_KEY', 'sk_gYgpi.JtX=o2qvFk2aj3RD+cEVU3u' );
define( 'WP__GRU_FS__LICENSE_KEY', 'sk_pM50Qt3.G:iz5NQc=X+@xb^gR&5ub' );


freemius-license-installer.php
<?php
/**
 * Freemius License Installer.
 *
 * @package freemius-license-installer
 * @author  Andy Fragen
 * @link    https://github.com/afragen/freemius-license-installer
 * @uses    https://github.com/squarecandy/freemius-auto-activation
 *
 * Plugin Name:       Freemius License Installer
 * Plugin URI:        https://github.com/afragen/freemius-license-installer
 * Description:       Add Freemius licensing from .env file.
 * Version:           1.1.0
 * Author:            Andy Fragen
 * License:           MIT
 * Requires at least: 5.9
 * Requires PHP:      7.2
 * GitHub Plugin URI: https://github.com/afragen/freemius-license-installer
 * Primary Branch:    main
 */

namespace Fragen\Freemius_License_Installer;

/*
 * Exit if called directly.
 * PHP version check and exit.
 */
if ( ! defined( 'WPINC' ) ) {
	die;
}

require_once __DIR__ . '/vendor/autoload.php';

add_action(
	'plugins_loaded',
	function() {
		$config = [
			[
				'name'     => 'Freemius Auto Activation',
				'host'     => 'github',
				'slug'     => 'freemius-auto-activation/freemius-auto-activation.php',
				'uri'      => 'https://github.com/squarecandy/freemius-auto-activation',
				'branch'   => 'main',
				'required' => true,
			],
		];
		\WP_Dependency_Installer::instance( __DIR__ )->register( $config )->run();
	}
);

// Load licenses from .env file.
( \Dotenv\Dotenv::createImmutable( dirname( __FILE__ ) ) )->load();

$fs_shortcodes = explode( ',', $_ENV['fs_shortcodes'] );

// Define constants for Freemius Auto Activation.
define( 'FS_SHORTCODES', $fs_shortcodes );
foreach ( $fs_shortcodes as $fs_shortcode ) {
	define( 'WP__' . strtoupper( $fs_shortcode ) . '__LICENSE_KEY', $_ENV[ $fs_shortcode ] );
}

register_deactivation_hook( __FILE__, __NAMESPACE__ . '\delete_freemius_auto_activation' );

/**
 * Upon deactivation, delete freemius-auto-activation plugin.
 *
 * @return void
 */
function delete_freemius_auto_activation() {
	global $wp_filesystem;

	if ( ! $wp_filesystem ) {
		require_once ABSPATH . '/wp-admin/includes/file.php';
		WP_Filesystem();
	}

	$wp_filesystem->delete( $wp_filesystem->wp_plugins_dir() . 'freemius-auto-activation', true );
}
